### PR TITLE
Support Web Redemption Links

### PR DESCRIPTION
**Pull request by tonidero**
This adds support for Redemption Links in PHC which will be used by the hybrids.

Related RN PR: https://github.com/RevenueCat/react-native-purchases/pull/1145

### CODE REVIEW

File: ios/PurchasesHybridCommon/PurchasesHybridCommon/CommonFunctionality.swift [hunk 1]
```diff
@@ -754,6 +754,56 @@ import RevenueCat
...
+
+    @objc(isWebPurchaseRedemptionURL:)
+    static func isWebPurchaseRedemptionURL(urlString: String) -> Bool {
+        guard let url = URL(string: urlString) else { return false }
+
+        return url.asWebPurchaseRedemption != nil
+    }
+
+    static func redeemWebPurchase(urlString: String,
+                                  completion: @escaping ([String: Any]?, ErrorContainer?) -> Void) {
```
Comment (JayShortway) [inline ios/PurchasesHybridCommon/PurchasesHybridCommon/CommonFunctionality.swift:768]: Just a question, do we not need this from ObjC? Or does it take the annotation from the `extension`?

File: android/hybridcommon/src/main/java/com/revenuecat/purchases/hybridcommon/common.kt [hunk 2]
```diff
@@ -600,8 +605,77 @@ fun getPromotionalOffer(): ErrorContainer {
...
+        is RedeemWebPurchaseListener.Result.Expired -> "EXPIRED"
+    }
+}
+
+@OptIn(ExperimentalPreviewRevenueCatPurchasesAPI::class)
+private fun String.toWebPurchaseRedemption(): WebPurchaseRedemption? {
+    try {
+        // Replace this with parseAsWebPurchaseRedemption overload
+        // accepting strings once it's available.
+        val intent = Intent(Intent.ACTION_VIEW, Uri.parse(this))
```
[the remark this answers is not in the document]
Comment (JayShortway) [reply android/hybridcommon/src/main/java/com/revenuecat/purchases/hybridcommon/common.kt:672]: Wouldn't be opposed to adding Robolectric to PHC (neither to making it KMP haha), but your call.

**Review by JayShortway — APPROVED** [review]
Looks good!

File: ios/PurchasesHybridCommon/PurchasesHybridCommon/CommonFunctionality.swift (hunk 1 above)
Comment (tonidero) [reply ios/PurchasesHybridCommon/PurchasesHybridCommon/CommonFunctionality.swift:768]: Good question... it worked when I tested it in RN, so it seems it's not needed... But going to add the annotation to be consistent. Thanks!

File: android/hybridcommon/src/main/java/com/revenuecat/purchases/hybridcommon/common.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (tonidero) [reply android/hybridcommon/src/main/java/com/revenuecat/purchases/hybridcommon/common.kt:672]: Yeah haha, will leave that to when we have some more time... But yeah, I think we probably should do something like that :P

**Pull request merged**